p2p-specs/p2p-interface.md

```diff
@@ -568,7 +587,7 @@ Request Content:
 Response Content: 
 ```
 (
-  List[UserOp, MAX_OPS_PER_REQUEST]
+  List[VerifiedUserOperation, MAX_OPS_PER_REQUEST]
```

Comment:
1. the text below says "request UserOp mempool for the given EntryPoint" - which is not given in the request
2. the response type "VerifiedUserOperation" carries "entrypoint" for each entry, which is useless (they are all expected to be the same)
3. (still, we do need the `verified_at_block_hash` for each UserOp)

Reply:
I can create a new container "PooledUserOp" without the entrypoint and add the entrypoint to the requests of both pooled_user_ops_by_hash and pooled_user_op_hashes.

Reply:
I think we should just remove the text below and return the entrypoint as part of `VerifiedUserOperation` in this response.

`pooled_user_ops_by_hash` doesn't need an entry point in the request because the hash is unique. The response can tell the requester what entry point the UO was verified against.

`pooled_user_op_hashes` similarly doesn't need an entry point in the request because the peer should just return hashes for all mempools that they have in common. After deduping the hashes the sender can then request what they don't know about already, and then learn what entry point/block the UO was verified against.

Reply:
I see three options:
1- keep the response for pooled_user_ops_by_hash as is where the entrypoint is included with every userOp.
2-change the response for pooled_user_ops_by_hash to Userops grouped by an entrypoint address.
3-add the entrypoint in the request to avoid including it in the response (my suggestion), and you would need to add it to pooled_user_op_hashes request also for this to work even though the hashes are unique.

Reply:
My vote would be 1 to limit the amount of times you need to call `pooled_user_ops_by_hash`. Just call it once for each peer during sync and receive a list of UO hashes to seed your mempool.